core/vtk/ttkRipsPersistenceDiagram/ttkRipsPersistenceDiagram.cpp

```diff
@@ -0,0 +1,209 @@
+                                               pair.second.first.end()));
+
+      ++i;
+    }
+  }
+
+  offsets->SetTuple1(n_pairs, connectivity->GetNumberOfTuples());
+
+  vtkNew<vtkCellArray> cells{};
+  cells->SetData(offsets, connectivity);
+  vtu->SetPoints(points);
+  vtu->SetCells(VTK_LINE, cells);
+
+  // add diagonal
+  std::array<vtkIdType, 2> diag{0, 2 * i_max};
+  vtu->InsertNextCell(VTK_LINE, 2, diag.data());
+  pairsId->InsertTuple1(n_pairs, -1);
+  pairsDim->InsertTuple1(n_pairs, -1);
+  isFinite->InsertTuple1(n_pairs, false);
+  persistence->InsertTuple1(n_pairs, 0.);
+  birthScalars->InsertTuple1(n_pairs, 0.);
+
+  return 1;
+}
+
+ttkRipsPersistenceDiagram::ttkRipsPersistenceDiagram() {
+  this->SetNumberOfInputPorts(1);
+  this->SetNumberOfOutputPorts(1);
+}
+
+int ttkRipsPersistenceDiagram::FillInputPortInformation(int port,
+                                                        vtkInformation *info) {
+  if(port == 0) {
+    info->Set(vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE(), "vtkTable");
+    return 1;
+  }
+  return 0;
+}
+
+int ttkRipsPersistenceDiagram::FillOutputPortInformation(int port,
+                                                         vtkInformation *info) {
+  if(port == 0) {
+    info->Set(vtkDataObject::DATA_TYPE_NAME(), "vtkUnstructuredGrid");
+    return 1;
+  }
+  return 0;
+}
+
+int ttkRipsPersistenceDiagram::RequestData(vtkInformation *ttkNotUsed(request),
+                                           vtkInformationVector **inputVector,
+                                           vtkInformationVector *outputVector) {
+
+  ttk::Timer tm{};
+
+  vtkTable *input = vtkTable::GetData(inputVector[0]);
+  vtkUnstructuredGrid *outputPersistenceDiagram
+    = vtkUnstructuredGrid::GetData(outputVector);
+
+  if(!input)
+    return 0;
+
+  std::vector<std::vector<double>> points;
+  if(!InputIsDistanceMatrix) {
+    const int numberOfPoints = input->GetNumberOfRows();
+    const int dimension = input->GetNumberOfColumns();
+
+    points = std::vector<std::vector<double>>(numberOfPoints);
+    for(int i = 0; i < numberOfPoints; ++i) {
+      for(int j = 0; j < dimension; ++j)
+        points[i].push_back(input->GetValue(i, j).ToDouble());
+    }
+    this->printMsg("Ripser starts (#dim: " + std::to_string(dimension)
+                     + ", #pts: " + std::to_string(numberOfPoints) + ")",
+                   1.0, tm.getElapsedTime(), 1);
+  } else {
+    const int n
+      = std::min(input->GetNumberOfRows(), input->GetNumberOfColumns());
+    const int column_offset = input->GetNumberOfColumns() - n;
+
+    points = {std::vector<double>(n * (n - 1) / 2)};
+    for(int i = 1; i < n; ++i) {
+      for(int j = 0; j < i; ++j)
+        points[0][i * (i - 1) / 2 + j]
+          = input->GetValue(i, j + column_offset).ToDouble();
+    }
+    this->printMsg("Ripser starts (" + std::to_string(n) + "x"
+                     + std::to_string(n) + " dist mat)",
+                   1.0, tm.getElapsedTime(), 1);
+  }
+
```

Comment:
I suggest to add a new output message (`printMsg`) here to remind the user parameters (`SimplexMaximumDimension` and `SimplexMaximumDiameter`)